src/briefcase/integrations/subprocess.py

```diff
@@ -469,19 +481,23 @@ def stream_output(self, label, popen_process, stop_func=lambda: False):
             # re-raise to exit as "Aborted by user"
             raise
         finally:
-            self.cleanup(label, popen_process)
-            streamer_deadline = time.time() + 3
-            while output_streamer.is_alive() and time.time() < streamer_deadline:
-                time.sleep(0.1)
-            if output_streamer.is_alive():
-                self.tools.logger.error(
-                    "Log stream hasn't terminated; log output may be corrupted."
-                )
+            if skip_cleanup:
+                output_streamer_stop_flag.set()
+            else:
+                self.cleanup(label, popen_process)
+                streamer_deadline = time.time() + 3
+                while output_streamer.is_alive() and time.time() < streamer_deadline:
+                    time.sleep(0.1)
+                if output_streamer.is_alive():
+                    self.tools.logger.error(
+                        "Log stream hasn't terminated; log output may be corrupted."
+                    )
```

Comment:
With the introduction of a "stop flag" for the output streamer, I think it begs the question if it should be used as well as an alternative to printing "log stream hasn't terminated".

Given we don't have intended use-cases for process termination to fail, I'm leaning towards not setting the stop flag to get as much output as possible from the process.